src/exporter/DaemonMetricCollector.cc

```diff
@@ -14,24 +16,25 @@ using json_object = boost::json::object;
 using json_value = boost::json::value;
 using json_array = boost::json::array;
 
-const char *DaemonMetricCollector::SOCKETDIR = "/var/run/ceph/";
-
 void DaemonMetricCollector::request_loop(boost::asio::steady_timer &timer) {
   timer.async_wait([&](const boost::system::error_code &e) {
     std::cerr << e << std::endl;
+    const auto start = std::chrono::system_clock::now();
     update_sockets();
     dump_asok_metrics();
-    timer.expires_from_now(std::chrono::seconds(stats_period));
+    const std::chrono::duration<double> end = std::chrono::system_clock::now() - start;
+    timer.expires_from_now(std::chrono::seconds(std::max(stats_period, (int)std::round(end.count()))));
```

Comment:
Well it's just max of `(stats_period , < time_taken(update sockets + updating metrics))`

Reply:
The reason I did this way bcz if the actual time taken was more than mentioned stats_period then refreshing it with given stats_period doesn't makes sense and if it's less than `stats_period` => it should refresh after `stats_period`, only then it makes sense to provide it as an argument, right?

Reply:
yeah, this is worth some more thought. if there's a long delay here, that *might* just be because one daemon is slow to respond. if the others are fast, then it doesn't make as much sense to poll them less

ultimately it might be better to process these sockets in parallel, so we can make progress on other things if a small subset of the daemons are slow to respond. but that really complicates the design, and it may end up being an issue. just something to keep in mind when testing this stuff